python/tskit/trees.py

```diff
@@ -2327,56 +2331,83 @@ def _node_edges(self):
             if index == self.index:
                 return node_edge_map
 
-    # TODO make this a bit less embarrassing by using an iterative method.
-    def __build_newick(self, *, node, precision, node_labels, include_branch_lengths):
+    def _as_newick_fast(self, *, root, precision, legacy_ms_labels):
         """
-        Simple recursive version of the newick generator used when non-default
-        node labels are needed, or when branch lengths are omitted
+        Call into the fast but limited C implementation of the newick conversion.
         """
-        label = node_labels.get(node, "")
-        if self.is_leaf(node):
-            s = f"{label}"
-        else:
-            s = "("
-            for child in self.children(node):
-                branch_length = self.branch_length(child)
-                subtree = self.__build_newick(
-                    node=child,
-                    precision=precision,
-                    node_labels=node_labels,
-                    include_branch_lengths=include_branch_lengths,
-                )
-                if include_branch_lengths:
-                    subtree += ":{0:.{1}f}".format(branch_length, precision)
-                s += subtree + ","
-            s = s[:-1] + f"){label}"
-        return s
+        root_time = max(1, self.time(root))
+        max_label_size = math.ceil(math.log10(self.tree_sequence.num_nodes))
+        single_node_size = (
+            5 + max_label_size + math.ceil(math.log10(root_time)) + precision
+        )
+        buffer_size = 1 + single_node_size * self.num_nodes
+        return self._ll_tree.get_newick(
+            precision=precision,
+            root=root,
+            buffer_size=buffer_size,
+            legacy_ms_labels=legacy_ms_labels,
+        )
+        return
```

Comment:
Looks like an extra return slipped in.

Reply:
Aha, thanks! Surprising flake8 didn't spot this.